provider/azure/environ.go

```diff
@@ -690,16 +689,20 @@ func (env *azureEnviron) createVirtualMachine(
 
 	publicIPAddressName := vmName + "-public-ip"
 	publicIPAddressId := fmt.Sprintf(`[resourceId('Microsoft.Network/publicIPAddresses', '%s')]`, publicIPAddressName)
+	publicIPAddressAllocationMethod := network.Static
+	if strings.ToLower(env.config.loadBalancerSkuName) == "basic" { // preserve the settings that were used in Juju 2.4 and earlier
```

Comment:
Since you're title-casing the setting when creating it, might make more sense to just compare to "Basic" here?